t/unit/worker/test_request.py

```diff
@@ -629,6 +632,7 @@ def test_on_hard_timeout(self, patching):
         job.acknowledge = Mock(name='ack')
         job.task.acks_late = True
         job.on_timeout(soft=False, timeout=1337)
+        self.app.set_current()
```

Comment:
Any idea why this is now required also? I am asking just to verify that test cases are still valid before/after the changes.

Reply:
Sure.  There was an error with the app not being set, and the default app being used. It triggered the "celery app trap" when ran with pytest.  I'm not sure why this cropped up all of a sudden though.

I'll revert this one too and let TravisCI run, and you can take a look.   Thanks!